Trie/Trie/Trie.cs

```diff
@@ -0,0 +1,165 @@
+namespace Trie;
+
+/// <summary>
+/// Trie, type of k-ary search tree used for storing and searching a specific key from a set.
+/// </summary>
+public class Trie
+{
+    private readonly TrieVertex root = new ();
+
+    /// <summary>
+    /// Gets number of words in trie.
+    /// </summary>
+    public int Size { get; private set; }
+
+    /// <summary>
+    /// Function that adds word to trie.
+    /// </summary>
+    /// <param name="element">Word that we adds.</param>
+    /// <returns>True if word wasn't in trie else false.</returns>
+    /// <exception cref="ArgumentNullException">Word can't be null.</exception>
+    public bool Add(string? element)
```

Comment:
Зачем делать nullable, если в комментарии пишете, что оно не может быть нуллом? Проверка имеет смысл даже без `?`, параметр в теории может прийти из проекта, в котором отключена nullable-нотация или вообще из другого языка на .NET,
```suggestion
    public bool Add(string element)
```